src/main/java/domain/AttendanceHistory.java

```diff
@@ -0,0 +1,121 @@
+
+    private final Crew crew;
+    private final Map<LocalDate, AttendanceRecord> attendanceHistory;
+
+    public AttendanceHistory(final Crew crew) {
+        this.attendanceHistory = new HashMap<>();
+        this.crew = crew;
+    }
+
+    public AttendanceRecord attendance(final LocalDateTime attendanceDateTime) {
+        validateAttendanceDay(attendanceDateTime);
+        validateAlreadyAttendance(attendanceDateTime);
+        final AttendanceRecord attendanceRecord = new AttendanceRecord(attendanceDateTime);
+        attendanceHistory.put(attendanceDateTime.toLocalDate(), attendanceRecord);
+        return attendanceRecord;
+    }
+
+    private void validateAlreadyAttendance(final LocalDateTime attendanceDateTime) {
+        if (attendanceHistory.containsKey(attendanceDateTime.toLocalDate())) {
+            throw new IllegalStateException("주어진 날짜는 이미 출석이 완료되었습니다.");
+        }
+    }
+
+    public boolean isAlreadyAttendance(final LocalDate targetDate) {
+        return attendanceHistory.containsKey(targetDate);
+    }
+
+    private void validateAttendanceDay(final LocalDateTime attendanceDateTime) {
+        if (!isAttendanceDay(attendanceDateTime.toLocalDate())) {
+            throw new IllegalArgumentException("주어진 날짜는 출석하는 날이 아닙니다.");
+        }
+    }
+
+    public AttendanceRecord updateTimeByDate(final LocalDateTime afterTime) {
+        validateAttendanceDay(afterTime);
+        validateAttendanceTime(afterTime);
+        final AttendanceRecord prevRecord = attendanceHistory.get(afterTime.toLocalDate());
+        final AttendanceRecord newRecord = new AttendanceRecord(afterTime);
+        attendanceHistory.put(afterTime.toLocalDate(), newRecord);
+        return prevRecord;
+    }
+
+    private static void validateAttendanceTime(final LocalDateTime afterTime) {
+        final LocalTime time = afterTime.toLocalTime();
+        if (time.isBefore(START_TIME) || time.isAfter(END_TIME)) {
+            throw new IllegalArgumentException("출석이 가능한 시간이 아닙니다.");
+        }
+    }
+
+    public AttendanceRecord findByDate(final LocalDate date) {
+        validateAttendanceDay(LocalDateTime.of(date, LocalTime.NOON));
+        if (!attendanceHistory.containsKey(date)) {
+            return AttendanceRecord.empty(date);
+        }
+        return attendanceHistory.get(date);
+    }
+
+    public List<AttendanceRecord> findAllUntilBeforeToday(final LocalDate targetDate) {
+        return ATTENDANCE_HISTORY_RECORD_BEGIN_DATE.datesUntil(targetDate)
+                .filter(this::isAttendanceDay)
+                .map(date -> attendanceHistory.getOrDefault(date, AttendanceRecord.empty(date)))
+                .collect(Collectors.toList());
+    }
+
+    private boolean isAttendanceDay(final LocalDate targetDate) {
+        if (DayOfWeeks.isWeekend(targetDate.getDayOfWeek())
+                || SolarCalendarHoliday.isHoliday(MonthDay.from(targetDate))) {
+            return false;
+        }
+        return true;
```

Comment:
boolean을 반환하는 타입은 아래와 같이 if 문 없이도 해결되는 경우가 꽤 많답니다. 😃 

```suggestion
        return !DayOfWeeks.isWeekend(targetDate.getDayOfWeek())
            && !SolarCalendarHoliday.isHoliday(MonthDay.from(targetDate));
```